internal/ethapi/api.go

```diff
@@ -1098,7 +1098,21 @@ func (s *PublicBlockChainAPI) GetDiffAccounts(ctx context.Context, blockNr rpc.B
 		return nil, fmt.Errorf("block not found for block number (%d): %v", blockNr, err)
 	}
 
-	return s.b.Chain().GetDiffAccounts(header.Hash())
+	accounts, err := s.b.Chain().GetDiffAccounts(header.Hash())
+	if err == nil || err.Error() != "no diff layer found" {
```

Comment:
Can we define this error in `core/error.go`, and compare it with `errors.Is(err, core.ErrDiffLayerNofFound)` instead of string comparison?